contracts/drafts/dao/DAO.sol

```diff
@@ -0,0 +1,117 @@
+    }
+
+    function restartFundingRound() public onlyOwner {
+        require(
+            currentState == "LIVE" || currentState == "FAILED",
+            "Initial funding round not ended."
+        );
+        _transition("SETUP");
+    }
+
+    function begMoneyForIdea(uint256 amount, address payable idea) public {
+        require(currentState == "LIVE", "Founders not defined yet.");
+        require(
+            totalAmounts + amount <= address(this).balance,
+            "You beg too much."
+        );
+        begAmount[idea] = amount;
+    }
+
+    function voteForIdea(uint256 votes, address idea) public {
+        require(currentState == "LIVE", "Founders not defined yet.");
+        require(
+            this.balanceOf(msg.sender) - totalVotesByHolder[msg.sender] >= votes,
+            "Not enough power."
+        );
+        totalVotesForIdea[idea] += votes;
+        totalVotesByHolder[msg.sender] += votes;
+        votesForIdeaByHolder[idea][msg.sender] += votes;
+        resolveBackerForIdea(idea, msg.sender);
+    }
+
+    function fundIdea(address payable idea) public {
+        require(currentState == "LIVE", "Founders not defined yet.");
+        require(
+            totalVotesForIdea[idea] >= this.totalSupply().div(2) + 1,
+            "Not enough expressed votes."
+        );
+        // solium-disable-next-line security/no-call-value
+        IssuanceEth(idea).invest.value(begAmount[idea])();
+        totalAmounts -= begAmount[idea];
+        for (uint256 i = 0; i < backersForIdea[idea].length; i++) {
+            totalVotesByHolder[backersForIdea[
+                    idea
+                ][i]] -= votesForIdeaByHolder[
+                    idea
+                ][backersForIdea[idea][i]];
+        }
+        delete totalVotesForIdea[idea];
+        delete begAmount[idea];
+    }
+
+    function getTokensForFundedIdea(address idea) public {
+        require(currentState == "LIVE", "Founders not defined yet.");
+        IssuanceEth issuance = IssuanceEth(idea);
+        ERC20Mintable issuanceToken = ERC20Mintable(
+            address(issuance.issuanceToken())
+        );
+        issuance.withdraw();
+    }
+
+    function getReturnsFromTokensOfFundedIdea(address idea) public {
+        require(currentState == "LIVE", "Founders not defined yet.");
+        IssuanceEth issuance = IssuanceEth(idea);
+        address dividendToken = address(issuance.issuanceToken());
+        uint256 tokenIndexBefore = tokenIndex;
+        resolveDividendToken(dividendToken);
+        require(tokenIndex != tokenIndexBefore, "Cannot get returns again.");
+        totalDividends[dividendToken] = totalDividends[dividendToken].add(
+                ERC20Mintable(dividendToken).balanceOf(address(this))
+            );
+        totalDividendPoints[dividendToken] = totalDividends[dividendToken]
+            .mul(pointMultiplier).div(this.totalSupply());
+    }
+
+    function resolveBackerForIdea(address idea, address backer) internal {
```

Comment:
Is `resolve` the right word? What does this do?

Reply:
I think it's the right word. It either returns void (in case backer is already resolved for idea) or it pushes it to the back of the backers for that idea array. It's basically an insert into a set (I guess I could use `insert` instead of `resolve`? That would have to be used also for `resolveDividendToken` in ERC20MultiDividendable.